logic.py

```diff
@@ -234,7 +236,7 @@ def vote(voter, proposal, scores, nominate=False):
 
     if set(scores.keys()) != set(x.id for x in get_standards()):
         return None
-    for v in scores.values():
+    for v in list(scores.values()):
```

Comment:
Is this necessary? You only need to go from iterator to list places where you will use list behavior. Here you are just for-looping so an iterator is fine.